### PR TITLE
Add index in target controller

### PR DESCRIPTION
**Pull request by antoalconti**
- The index was added to get user's targets.

- Index spect was added.

### CODE REVIEW

File: spec/requests/api/v1/targets/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe 'GET /api/v1/Targets', type: :request do
+  subject { get api_v1_targets_url, headers: headers, as: :json }
+
+  let!(:user) { create(:user) }
+  let(:topic) { create(:topic) }
```
Comment (santir489) [inline spec/requests/api/v1/targets/index_spec.rb]: `!`?

File: spec/requests/api/v1/targets/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  context 'with logged-in user' do
+    let(:headers) { auth_headers }
+
+    context 'when the request is valid and the user has targets' do
+      let!(:targets) { create_list(:target, 5, user: user, topic: topic) }
```
Comment (santir489) [inline spec/requests/api/v1/targets/index_spec.rb]: How can you be sure you are returning only the current user target? And not all targets? I would create some targets that does not belongs to the current user to check that.

File: spec/requests/api/v1/targets/index_spec.rb (hunk 1 above)
Comment (antoalconti) [reply spec/requests/api/v1/targets/index_spec.rb]: It was not necessary,  thank you.

File: spec/requests/api/v1/targets/index_spec.rb (hunk 2 above)
Comment (antoalconti) [reply spec/requests/api/v1/targets/index_spec.rb]: I did not realize to check it, will test now thank you.

File: spec/requests/api/v1/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  let(:user) { create(:user) }
+  let(:other_user) { create(:user) }
+  let(:topic) { create(:topic) }
+
+  context 'with logged-in user' do
+    let(:headers) { auth_headers }
+
+    context 'when the request is valid and the user has targets' do
+      let!(:targets) { create_list(:target, 5, user: user, topic: topic) }
+      let!(:targets_aux) { create_list(:target, 4, user: other_user, topic: topic) }
```
Comment (santir489) [inline spec/requests/api/v1/targets/index_spec.rb]: I would prefer to use other word than aux.. like  `others_targets `

**Pull request merged**